core/modules/qdisp/JobQuery.cc

```diff
@@ -95,25 +100,38 @@ void JobQuery::provisioningFailed(std::string const& msg, int code) {
 
 /// Cancel response handling. Return true if this is the first time cancel has been called.
 bool JobQuery::cancel() {
+    LOGF_DEBUG("%1% JobQuery::cancel()" % getIdStr());
     if (_cancelled.exchange(true) == false) {
         std::lock_guard<std::recursive_mutex> lock(_rmutex);
-        // Nothing needs to be done for _queryResourcePtr.
-        if (_queryRequestPtr) {
+        // If _queryRequestPtr is not nullptr, then this job has been passed to xrootd and cancellation is complicated.
+        if (_queryRequestPtr != nullptr) {
```

Comment:
I believe that `if (_queryRequestPtr)` convention is much more common in our code than `if (_queryRequestPtr != nullptr)`.